tests/wpt/servowpt.py

```diff
@@ -186,41 +188,106 @@ def update_tests(**kwargs):
     return 1 if return_value is update.exit_unclean else 0
 
 
-class TrackerFilter():
+class GithubContextInformation(NamedTuple):
+    build_url: str
+    pull_url: str
+    branch_name: str
+
+
+class TrackerDashboardFilter():
     def __init__(self):
-        self.url = os.environ.get(TRACKER_API_ENV_VAR, TRACKER_API)
-        if self.url.endswith("/"):
-            self.url = self.url[0:-1]
+        base_url = os.environ.get(TRACKER_API_ENV_VAR, TRACKER_API)
+        self.headers = {
+            "Content-Type": "application/json"
+        }
+        if TRACKER_DASHBOARD_SECRET_ENV_VAR in os.environ:
+            self.url = f"{base_url}/dashboard/attempts"
+            secret = os.environ[TRACKER_DASHBOARD_SECRET_ENV_VAR]
+            self.headers["Authorization"] = f"Bearer {secret}"
+        else:
+            self.url = f"{base_url}/dashboard/query"
 
-    def is_failure_intermittent(self, test_name):
-        query = urllib.parse.quote(test_name, safe='')
-        request = urllib.request.Request("%s/query.py?name=%s" % (self.url, query))
-        search = urllib.request.urlopen(request)
-        return len(json.load(search)) > 0
+    @staticmethod
+    def get_github_context_information() -> GithubContextInformation:
+        github_context = json.loads(os.environ.get("GITHUB_CONTEXT", "{}"))
+        if not github_context:
+            return GithubContextInformation("", "", "")
 
+        repository = github_context['repository']
+        repo_url = f"https://github.com/{repository}"
 
-class GitHubQueryFilter():
-    def __init__(self, token):
-        self.token = token
+        run_id = github_context['run_id']
+        build_url = f"{repo_url}/actions/runs/{run_id})"
```

Comment:
i was thinking ideally we would have the full job build url, e.g.

https://github.com/servo/servo/actions/runs/4181616898/jobs/7244211010

so you can go from the dashboard directly to the chunk that failed, and view the failed run even if someone reruns the job in place.

but i can’t see the job build id [anywhere in the context docs](https://docs.github.com/en/actions/learn-github-actions/contexts), and we rarely rerun jobs in place, and having the full job build url weakens the “all results have the same” feature anyway… so on second thought, i think this approach is the way to go :)

Reply:
Oh, this is a good point. I was thinking about this today. Sometimes it would be nice to know what chunk something failed in (when referring back from the [job that reports aggregated results for instance](https://github.com/servo/servo/actions/runs/4182416519/jobs/7246803223)).